### PR TITLE
Implement isMaster with tests for replica and non-replica scenario

### LINKED ISSUE #59: Add isMaster command
https://docs.mongodb.com/v4.0/reference/command/isMaster/

### PR DESCRIPTION
**Pull request by tinchodias**
Fixes #59 plus few secondary additions:

* Add asMongoUrl (in String and ZnURL), which was already referenced from Mongo tests package but it's provided by Voyage-Mongo (if this PR is merged then I will remove it from there)
* Add Mongo>>closeIfOpen for convenience

### CODE REVIEW

File: mc/Mongo-Core.package/MongoIsMaster.class/instance/initializeWith..st [hunk 1]
```diff
@@ -0,0 +1,5 @@
+initialization
+initializeWith: aCollection
```
Comment (zecke) [inline mc/Mongo-Core.package/MongoIsMaster.class/instance/initializeWith..st]: Is it a collection or dictionary?

File: mc/Mongo-Core.package/MongoIsMaster.class/instance/urlString.st [hunk 2]
```diff
@@ -0,0 +1,5 @@
+replica set
+urlString
```
Comment (zecke) [inline mc/Mongo-Core.package/MongoIsMaster.class/instance/urlString.st]: How is this used? Can you think of a better name?

**Comment by zecke** [comment]
Looks good to me.

File: mc/Mongo-Core.package/MongoDatabase.class/instance/isMaster.st [hunk 3]
```diff
@@ -0,0 +1,7 @@
+operations
+isMaster
+
+	| reply |
+	reply := self command: (OrderedDictionary new at: #ismaster put: 1; yourself).
```
Comment (zecke) [inline mc/Mongo-Core.package/MongoDatabase.class/instance/isMaster.st]: What's the reason to use OrderedIdentityDictionary in many other places?

File: mc/Mongo-Core.package/MongoIsMaster.class/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+isMaster returns a document that describes the role of the mongod instance. If the optional field saslSupportedMechs is specified, the command also returns an array of SASL mechanisms used to create the specified user’s credentials.
+
+If the instance is a member of a replica set, then isMaster returns a subset of the replica set configuration and status including whether or not the instance is the primary of the replica set.
+
+When sent to a mongod instance that is not a member of a replica set, isMaster returns a subset of this information.
+
+MongoDB drivers and clients use isMaster to determine the state of the replica set members and to discover additional members of a replica set.
+
+
+Read more at: https://docs.mongodb.com/v4.0/reference/command/isMaster/
```
Comment (zecke) [inline mc/Mongo-Core.package/MongoIsMaster.class/README.md]: In the other PR we are linking to the most recent docs. It's fine to link to the doc used during impl or  latest but we should aim to be consistent.

File: mc/Mongo-Core.package/MongoIsMaster.class/instance/urlString.st (hunk 2 above)
Comment (tinchodias) [reply mc/Mongo-Core.package/MongoIsMaster.class/instance/urlString.st]: The intent was to differenciate from the cases when a ZnUrl is useful vs this string version of the url that always comes in the responses. May be `host`?

File: mc/Mongo-Core.package/MongoDatabase.class/instance/isMaster.st (hunk 3 above)
Comment (tinchodias) [reply mc/Mongo-Core.package/MongoDatabase.class/instance/isMaster.st]: Frankly I don't know, a reason may be to allow repeated String keys but I don't know in which special cases. I think we should unify how the "JSON dictionaries" (Dictionaries that will be converted to json in a mongo call) are created.

File: mc/Mongo-Core.package/MongoIsMaster.class/README.md (hunk 4 above)
Comment (tinchodias) [reply mc/Mongo-Core.package/MongoIsMaster.class/README.md]: Good catch. What do you think is the best approach?

File: mc/Mongo-Core.package/MongoIsMaster.class/instance/initializeWith..st (hunk 1 above)
Comment (tinchodias) [reply mc/Mongo-Core.package/MongoIsMaster.class/instance/initializeWith..st]: More particularly a dictionary, but when somebody presses on "create" in the debugger (in a DNU), the parameter has this name. This was defined in some method of the hierarchy, not sure if it's on purpose or a bug.

**Comment by tinchodias** [comment]
Do you consider it's good overall?

File: mc/Mongo-Core.package/MongoIsMaster.class/README.md (hunk 4 above)
Comment (zecke) [reply mc/Mongo-Core.package/MongoIsMaster.class/README.md]: Link to the document at the time of the implementation. One can go to newer versions (if it exists) then but not the other way around.

**Comment by zecke** [comment]
Yes. I consider this good over all. I don't have the power to merge it though. ;)

File: mc/Mongo-Core.package/MongoDatabase.class/instance/isMaster.st (hunk 3 above)
Comment (tinchodias) [reply mc/Mongo-Core.package/MongoDatabase.class/instance/isMaster.st]: I created issue #61 to maybe discuss more about this.

File: mc/Mongo-Core.package/MongoIsMaster.class/README.md (hunk 4 above)
Comment (tinchodias) [reply mc/Mongo-Core.package/MongoIsMaster.class/README.md]: I will take it into account for next time.

**Pull request merged**